src/primitives/transaction.cpp

```diff
@@ -123,7 +123,7 @@ std::string CTxIn::ToString() const
     if (prevout.IsNull())
         str += strprintf(", coinbase %s", HexStr(scriptSig));
     else
-        str += strprintf(", scriptSig=%s", scriptSig.ToString().substr(0,24));
+        str += strprintf(", scriptSig=%s", HexStr(scriptSig).substr(0, 24));
```

Comment:
wondering if there was a reason HexStr wasn't used here before

Reply:
Asked about it on bitcoin, seems fine.https://github.com/bitcoin/bitcoin/pull/5264#discussion_r182417714